mem0/vector_stores/milvus.py

```diff
@@ -0,0 +1,229 @@
+from pydantic import BaseModel
+from typing import Optional, Dict
+from mem0.configs.vector_stores.milvus import MetricType
+from pymilvus import MilvusClient, CollectionSchema, FieldSchema, DataType
+
+
+class OutputData(BaseModel):
+    id: Optional[str]  # memory id
+    score: Optional[float]  # distance
+    payload: Optional[Dict]  # metadata
+
+
+
+class MilvusDB():
+    def __init__(self, url: str, token: str, collection_name: str, embedding_model_dims: int, metric_type: MetricType) -> None:
+        """Initialize the MilvusDB database.
+
+        Args:
+            url (str): Full URL for Milvus/Zilliz server.
+            token (str): Token/api_key for Milvus/Zilliz server.
+            collection_name (str): Name of the collection (defaults to mem0).
+            embedding_model_dims (int): Dimensions of the embedding model (defaults to 1536).
+            metric_type (MetricType): Metric type for similarity search (defaults to L2).
+        """
+        
+        self.collection_name = collection_name
+        self.embedding_model_dims = embedding_model_dims
+        self.metric_type = metric_type
+        
+        self.client = MilvusClient(uri=url,token=token)
+        
+        self.create_col(
+            collection_name=self.collection_name,
+            vector_size=self.embedding_model_dims,
+            metric_type=self.metric_type
+        )
+       
+        
+    def create_col(
+        self, collection_name : str, vector_size : str, metric_type : MetricType = MetricType.COSINE
+    ) -> None:
+        """Create a new collection with index_type IVF_FLAT.
+
+        Args:
+            collection_name (str): Name of the collection (defaults to mem0).
+            vector_size (str): Dimensions of the embedding model (defaults to 1536).
+            metric_type (MetricType, optional): etric type for similarity search. Defaults to MetricType.COSINE.
+        """
+
+        if self.client.has_collection(collection_name):
+            print(f"Collection {collection_name} already exists. Skipping creation.")
+        else:
+            fields = [
+                FieldSchema(name="id", dtype=DataType.VARCHAR, is_primary=True, max_length=512),
+                FieldSchema(name="vectors", dtype=DataType.FLOAT_VECTOR, dim=vector_size),
+                FieldSchema(name="metadata", dtype=DataType.JSON),
+            ]
+            
+            schema = CollectionSchema(fields, enable_dynamic_field=True)
+
+            index = self.client.prepare_index_params(
+                field_name="vectors",
+                metric_type=metric_type,
+                index_type="IVF_FLAT",
```

Comment:
maybe "AUTOINDEX" is better , because milvus lite only support 'AUTOINDEX', 'HNSW', 'FLAT' so far. If users are using milvus lite with `IVF_FLAT` index and the `./milvus.db` as uri, it will raise an error.